src/Microsoft.Identity.Client/Cache/Items/MsalAppMetadataCacheItem.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Cache.Items
+{
+    /// <summary>
+    /// Apps shouldn't rely on its presence, unless the app itself wrote it. It means that SDK should translate absense of app metadata to the default values of its required fields.
+    /// Other apps that don't support app metadata should never remove existing app metadata.
+    /// App metadata is a non-removable entity.It means there's no need for a public API to remove app metadata, and it shouldn't be removed when removeAccount is called.
+    /// App metadata is a non-secret entity. It means that it cannot store any secret information, like tokens, nor PII, like username etc.
+    /// App metadata can be extended by adding additional fields when required.Absense of any non-required field should translate to default values for those field.
+    /// </summary>
+    [DataContract]
```

Comment:
I don't think you need these to be DataContract since you should serialize/deserialize the same way as the other objects (per-field) to support back/forwards compat.